plugins/Services/test.py

```diff
@@ -105,6 +106,69 @@ def testRegisterNoExperimentalExtensions(self):
         self.assertIsNone(self.irc.takeMsg())
 
 
+class JoinsBeforeIdentifiedTestCase(PluginTestCase):
+    plugins = ('Services',)
+    config = {
+        'plugins.Services.noJoinsUntilIdentified': False,
+    }
+    timeout = 0.1
+
+    def testSingleNetwork(self):
+        queuedJoin = ircmsgs.join('#test', prefix=self.prefix)
+        self.irc.queueMsg(queuedJoin)
+        self.assertEquals(self.irc.takeMsg(), queuedJoin, 'Join request did not go through.')
+
+
+class NoJoinsUntilIdentifiedTestCase(PluginTestCase):
+    plugins = ('Services',)
+    config = {
+        'plugins.Services.noJoinsUntilIdentified': True,
+    }
+    timeout = 0.1
+
+    def _identify(self, irc):
+        irc.feedMsg(IrcMsg(command='376', args=(self.nick,)))
+        msg = irc.takeMsg()
+        self.assertEquals(msg.command, 'PRIVMSG')
+        self.assertEquals(msg.args[0], 'NickServ')
+        irc.feedMsg(ircmsgs.notice(self.nick, 'now identified', prefix='NickServ'))
+
+    def testSingleNetwork(self):
+        try:
+            self.assertNotError('services password %s %s' % (self.nick, 'secret'))
+            queuedJoin = ircmsgs.join('#test', prefix=self.prefix)
+            self.irc.queueMsg(queuedJoin)
+            self.assertIsNone(self.irc.takeMsg(), 'Join message went through.')
+            self._identify(self.irc)
+            self.assertEquals(self.irc.takeMsg(), queuedJoin, 'Join request did not go through after identification.')
+        finally:
+            self.assertNotError('services password %s ""' % self.nick)
+
+    def testMultipleNetworks(self):
+        try:
+            net1 = copy(self)
+            net1.irc = getTestIrc('testnet1')
+            net1.assertNotError('services password %s %s' % (self.nick, 'secret'))
+            try:
```

Comment:
You can probably do this with a single `try` block

Reply:
How to make sure both of the finally statements are attempted though?

Reply:
I guess the strictly correct behavior isn't worth the decrease in readability.  I merged the two blocks.